### ISSUE TITLE
Add an "auto delete" capability for credential and proof exchange records

### ISSUE THREAD

**Issue by tipusinghaw** [openwallet-foundation/credo-ts#2123]
Add the ability to "auto-delete" credentials and proof exchange records. using the global configurations and method-level parameters.
that might help prevent the PII data and help with storage optimization.

**Comment by TimoGlastra** [comment]
Yes I agree, that's a good one, and could be done immediately after emitting the error/success event. However this won't help for exchanges that are never completed.

You can quite easily build this on top of credo. Once you receive the completed/error event you delete the record. You can also run a cron script that deletes all records older than e.g. an hour every x minutes/hours/days. 

We're working on some new models for storage that would also make it easy to find records that were created e.g. more than one hour ago.
